src/plot_api/helpers.js

```diff
@@ -114,7 +114,7 @@ exports.cleanLayout = function(layout) {
                 scene.camera = {
                     eye: {x: eye[0], y: eye[1], z: eye[2]},
                     center: {x: center[0], y: center[1], z: center[2]},
-                    up: {x: mat[1], y: mat[5], z: mat[9]}
+                    up: {x: 0, y: 0, z: 1} // we just ignore calculating camera z up in this case
```

Comment:
Previous calculations of `camera.up` for `gl3d_surface-lighting` & `gl3d_ibm-plot` didn't set up vector close to [0,0,1].
I thought we may ignore camera z up if `cameraposition` depreciated setting is used.

Reply:
Nice solution.